smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptWriter.java

```diff
@@ -216,7 +217,11 @@ boolean writeShapeDocs(Shape shape) {
         return shape.getTrait(DocumentationTrait.class)
                 .map(DocumentationTrait::getValue)
                 .map(docs -> {
-                    writeDocs(docs);
+                    if (shape.getTrait(DeprecatedTrait.class).isPresent()) {
```

Comment:
This can be simplified in both places:

```java
if ({condition}) {
    docs = "@deprecated\n\n" + docs;
}
writeDocs(docs);
```

Reply:
Incorporated in [`b5f8d21` (#258)](https://github.com/awslabs/smithy-typescript/pull/258/commits/b5f8d217d5c109b1abe35bc3620e6c0abb2482b6)